pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -208,6 +215,34 @@ func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile,
 	return matchingSeeds, nil
 }
 
+// filterMultiZonalSeeds filters seeds based on label, if shoot requires a multi-zonal control plane
+func filterMultiZonalSeeds(seedList []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) ([]gardencorev1beta1.Seed, error) {
```

Comment:
the name of the method indicates that it only filters multi-zonal seeds but it can either return a slice of multi-zonal seeds or non-multizonal seeds depending upon feature gate and annotation. It should ideally be renamed to `filterCandidateSeeds` or `filterSeedsMatchingMultiZonalAttribute`

Reply:
Yes, will change that. Thanks